### PR TITLE
Support content with doctype

### PR DESCRIPTION
**Pull request by kaisermann**

### CODE REVIEW

File: parser.js [hunk 1]
```diff
@@ -1,137 +1,161 @@
...
 
-  function parseTag () {
-    if (html.substr(0, 4) === '<!--') { // comments
-      parseComment();
+  function parseTag() {
+    if (html.substr(0, 4) === '<!--') {
+      // comments
+      parseComment()
+    } else if (doctypePattern.test(html)) {
+      parseDoctype()
```
Comment (kaisermann) [inline parser.js:50]: Added this

File: parser.js [hunk 2]
```diff
@@ -1,137 +1,161 @@
...
+      parseEdge(rstart, parseStartTag)
     }
-    parseTagDecode();
+    parseTagDecode()
+  }
+
+  function parseDoctype() {
+    var match = html.match(doctypePattern)
+    chars = false
+    html = html.substring(match[0].length)
```
Comment (kaisermann) [inline parser.js:62]: And this

File: test/insane.js [hunk 3]
```diff
@@ -1,182 +1,324 @@
...
+  t.end()
+})
+
+test('removes doctype', (t) => {
+  const input = `<!DOCTYPE html><div>foo</div>`
+  const output = `<div>foo</div>`
+
+  t.equal(insane(input), output)
+  t.end()
+})
```
Comment (kaisermann) [inline test/insane.js:324]: And this one

**Review by klzns — APPROVED** [review]

File: test/insane.js (hunk 3 above)
Comment (klzns) [reply test/insane.js:324]: nice!

**Pull request merged**